### PR TITLE
Scene

### PR DESCRIPTION
**Pull request by hollasch**
In the end, the changes needed to fix the various images in _The Next Week_ were fairly straight-forward, and a formal `scene` class was not needed. This is particularly true because only the second book's code has multiple scenes included in it.

Instead, I opted for a simple switch statement that sets up the viewing parameters and background color in addition to the prior geometry creation.

Leading up to this, I took the liberty to perform some clarifying refactorings as well.

### CODE REVIEW

File: src/TheRestOfYourLife/main.cc [hunk 1]
```diff
@@ -91,33 +91,32 @@ hittable_list cornell_box(camera& cam, double aspect) {
...
+        for (int i = 0; i < image_width; ++i) {
             vec3 color;
-            for (int s = 0; s < num_samples; ++s) {
-                auto u = (i + random_double()) / nx;
-                auto v = (j + random_double()) / ny;
+            for (int s = 0; s < samples_per_pixel; ++s) {
+                auto u = (i + random_double()) / image_width;
+                auto v = (j + random_double()) / image_height;
                 ray r = cam.get_ray(u, v);
                 color += ray_color(r, world, lights, max_depth);
```
Comment (trevordblack) [inline src/TheRestOfYourLife/main.cc]: Needs `background_color`.

**Comment by hollasch** [comment]
I did find the original background color of (0.7, 0.8, 1.0), but that was quite blue-shifted compared to the images in the book. I ultimately went for the less saturated blueish-white color, which seems to match the images better.

File: src/TheRestOfYourLife/main.cc (hunk 1 above)
Comment (hollasch) [reply src/TheRestOfYourLife/main.cc]: Done.

**Comment by trevordblack** [comment]
Everything looks good to me.
This turned into a bit more of a headache than you were probably imagining.

Thanks for the follow-through.

**Comment by hollasch** [comment]
Actually, much _less_ of a headache. I was planning on creating a new `scene` class, perhaps with a `render()` or `ray_color()` method, or perhaps a `scene` class and a `renderer` class, coupled with some data tables for each scene, blah blah blah.

It would probably be the correct thing to do for a released program, but for this project, it would be gratuitous overhead.

**Pull request merged**